### PR TITLE
Feature/7866 dcc schema 1.3

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
Upgrade certificate access to 1.3.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-7866

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/HealthCertifiedPerson/HealthCertifiedPersonViewModel.swift [hunk 1]
```diff
@@ -222,15 +222,14 @@ final class HealthCertifiedPersonViewModel {
...
-			let dateOfBirthDate = ISO8601DateFormatter.justLocalDateFormatter.date(from: dateOfBirthString)
+			let dateOfBirthString = healthCertifiedPerson.dateOfBirth
 		else {
 			return nil
 		}
 
 		return String(
 			format: AppStrings.HealthCertificate.Person.dateOfBirth,
-			DateFormatter.localizedString(from: dateOfBirthDate, dateStyle: .medium, timeStyle: .none)
+			DCCDateStringFormatter.formatedString(from: dateOfBirthString)
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/HealthCertifiedPerson/HealthCertifiedPersonViewModel.swift:232]: If I can trust Figma, this one should stay in a localized format. I think the new formatting rules only apply to the `HealthCertificateViewModel`

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/HealthCertificate/HealthCertificateViewModel.swift [hunk 2]
```diff
@@ -309,13 +309,10 @@ final class HealthCertificateViewModel {
...
-		if let dateOfBirthDate = healthCertificate.dateOfBirthDate {
-			dateOfBirthCellViewModel = HealthCertificateKeyValueCellViewModel(
-				key: "Geburtsdatum / Date of Birth (YYYY-MM-DD)",
-				value: ISO8601DateFormatter.justLocalDateFormatter.string(from: dateOfBirthDate)
-			)
-		}
+		let dateOfBirthCellViewModel = HealthCertificateKeyValueCellViewModel(
+			key: "Geburtsdatum / Date of Birth (YYYY-MM-DD)",
+			value: DCCDateStringFormatter.formatedString(from: healthCertificate.dateOfBirth)
+		)
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/HealthCertificate/HealthCertificateViewModel.swift:315]: I think we need to also apply this to the date of vaccination and the date of birth in the vaccination details (and the recovery details, but they are not merged yet: #2963)

**Review by pascal-brause — APPROVED** [review]

**Review by nickguendling — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/HealthCertifiedPerson/HealthCertifiedPersonViewModel.swift (hunk 1 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/HealthCertifiedPerson/HealthCertifiedPersonViewModel.swift:232]: Then we need a way to parse and localize something like "1986" or "1986-11".
I'll create a task for that.

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/HealthCertificate/HealthCertificateViewModel.swift (hunk 2 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/HealthCertificate/HealthCertificateViewModel.swift:315]: Right. I'll create a task.

Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/HealthCertificate/HealthCertificateViewModel.swift:315]: https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-7988

Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/HealthCertificate/HealthCertificateViewModel.swift:309]: "I think we need to also apply this to the date of vaccination and the date of birth in the vaccination details"

Done!

**Review by nickguendling — APPROVED** [review]

**Review by naveeddotio — APPROVED** [review]

**Pull request merged**